dace/transformation/interstate/condition_map_interchange.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+"""Interchange conditional blocks with nested map regions."""
+
+from dace import sdfg as sd
+from dace import dtypes
+from dace.sdfg import utils as sdutil
+from dace.sdfg.state import ControlFlowRegion, ConditionalBlock
+from dace.properties import CodeBlock
+from dace.sdfg.nodes import MapEntry, MapExit, NestedSDFG
+from dace.memlet import Memlet
+from dace.transformation import transformation
+from dace.sdfg.sdfg import InterstateEdge
+from dace.sdfg.utils import set_nested_sdfg_parent_references
+import copy
+
+
+@transformation.explicit_cf_compatible
+class ConditionMapInterchange(transformation.MultiStateTransformation):
+    """
+    If one or multiple maps are surrounded by a conditional block, moves the conditional block into the nested SDFG of maps.
+    """
+
+    cond_block = transformation.PatternNode(ConditionalBlock)
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.cond_block)]
+
+    def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
+        # The conditional block should only contain one condition
+        # Empty else branches can be preprocessed away
+        if len(self.cond_block.branches) != 1:
+            return False
+        branch: ControlFlowRegion = self.cond_block.branches[0][1]
+
+        # Each state in the branch is either empty or only contains maps
+        for state in branch.all_states():
+            for node in state.nodes():
+                if (not isinstance(node, (MapEntry, MapExit)) and state.entry_node(node) is None and any([
+                        not isinstance(n, (MapEntry, MapExit)) for n in set(state.successors(node))
+                        | set(state.predecessors(node))
+                ])):
+                    return False
+
+        return True
+
+    def apply(self, graph: ControlFlowRegion, sdfg: sd.SDFG):
+        branch: ControlFlowRegion = self.cond_block.branches[0][1]
+        branch_cond = self.cond_block.branches[0][0]
+        cond_syms = set(branch_cond.get_free_symbols())
+        all_states = list(branch.all_states())
+
+        # Prepend the condition computation
+        cond_sym = graph.sdfg.add_symbol("cond", dtypes.bool, find_new_name=True)
```

Comment:
Good that `find_new_name` is `True`, but please make the starting name a bit more specific to the conditional block it corresponds to. Maybe concat with `'_' + self.cond_block.label` or something.